src/state/routing/useRoutingAPITrade.ts

```diff
@@ -45,17 +47,34 @@ export function useRoutingAPITrade<TTradeType extends TradeType>(
     routerPreference,
   })
 
+  const shouldUseRoutingApiV2 = useRoutingAPIV2Enabled()
+
   const {
-    isError,
-    data: tradeResult,
-    currentData: currentTradeResult,
-  } = useGetQuoteQuery(queryArgs ?? skipToken, {
+    isError: isLegacyAPIError,
+    data: legacyAPITradeResult,
+    currentData: currentLegacyAPITradeResult,
+  } = useGetQuoteQuery(!shouldUseRoutingApiV2 ? queryArgs ?? skipToken : skipToken, {
     // Price-fetching is informational and costly, so it's done less frequently.
     pollingInterval: routerPreference === INTERNAL_ROUTER_PREFERENCE_PRICE ? ms`1m` : AVERAGE_L1_BLOCK_TIME,
     // If latest quote from cache was fetched > 2m ago, instantly repoll for another instead of waiting for next poll period
     refetchOnMountOrArgChange: 2 * 60,
   })
 
+  const {
+    isError: isV2APIError,
+    data: v2TradeResult,
+    currentData: currentV2TradeResult,
+  } = useGetQuoteQueryV2(shouldUseRoutingApiV2 ? queryArgs ?? skipToken : skipToken, {
+    // Price-fetching is informational and costly, so it's done less frequently.
+    pollingInterval: routerPreference === INTERNAL_ROUTER_PREFERENCE_PRICE ? ms`1m` : AVERAGE_L1_BLOCK_TIME,
```

Comment:
these look repeated from above?

Reply:
yup, copy pasted so it's easier to rip out the first block later

Reply:
i think just adding a comment to stuff you expect to rip out is useful context - don't worry about it this time but just helpful in the future 😄